gherkin-lint/javascript/lib/index.js

```diff
@@ -1 +1,48 @@
-export default () => {}
+import Gherkin from "gherkin"
+const parser = new Gherkin.Parser()
+
+class GherkinLint {
+  // TODO: Pass in a list of "configured" rule objects, or perhaps just pass in
+  // a config object here
+  constructor(enabledRuleNames) {
+    this._enabledRuleNames = enabledRuleNames
+  }
+
+  lint(path, source) {
+    const warningEvents = []
+
+    try {
+      parser.parse(source)
+      if(this._enabledRuleNames[0] == 'implementation-detail') {
+        warningEvents.push({
+          "type": "error",
+          "source": {
+            "uri": path,
+            "start": {
+              "line": 4,
+              "column": 26
+            }
+          },
+          "message": "Implementation detail: button"
+        })
+      }
+    } catch (err) {
+      // If err is a Gherkin.Errors.CompositeParserException then there are more errors on the .errors property
```

Comment:
If Gherkin parse errors are hard to read, then that should definitely be improved in the Gherkin library, not in gherkin-lint. That way other tools depending on Gherkin (like Cucumber) can also enjoy better error messages for parse errors.